addon/PageScraper.js

```diff
@@ -61,6 +62,28 @@ PageScraper.prototype = {
   }),
 
   /**
+   * Locally compute the size of the preview image
+   */
+  _computeImageSize(metadata) {
+    return new Promise(resolve => {
+      if (metadata.images[0]) {
```

Comment:
Check here for the array length and resolve and return fast, otherwise carry on and alias the image like

const metadataImage = metadata.images[0];

and refer to that